operator/iamkeycontroller/create.go

```diff
@@ -162,7 +162,7 @@ func (p *IAMKeyPipeline) createCredentialsSecret(ctx *pipelineContext) error {
 		for k, v := range connDetails {
 			secret.Data[k] = v
 		}
-		secret.Immutable = pointer.Bool(true)
+		secret.Immutable = ptr.To[bool](true)
```

Comment:
```suggestion
		secret.Immutable = ptr.To(true)
```

Generics can infer the type, no need to specify it.